generics/orderedmap/orderedmap.go

```diff
@@ -84,6 +84,16 @@ func (orderedMap *OrderedMap[K, V]) Delete(key K) bool {
 	return orderedMap.OrderedMap.Delete(key)
 }
 
+// Clone returns a copy of the orderedMap.
+func (orderedMap *OrderedMap[K, V]) Clone() (cloned *OrderedMap[K, V]) {
+	cloned = New[K, V]()
+	orderedMap.OrderedMap.ForEach(func(key, value interface{}) bool {
+		cloned.Set(key.(K), value.(V))
+		return true
+	})
```

Comment:
```suggestion
	orderedMap.ForEach(func(key K, value V) bool {
		cloned.Set(key, value)
		return true
	})
```

Would it work like this?

Reply:
I've tried, but ForEach receives a func with `interface{}` as parameters. We could make it generic, but I don't think it's worth it.